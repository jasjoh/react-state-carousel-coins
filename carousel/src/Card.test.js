import { render, fireEvent } from "@testing-library/react";
import Card from "./Card";
import TEST_IMAGES from "./_testCommon.js";

const title = "Test Title";

// Card Component Signature: Card({caption, src, currNum, totalNum})

it("Smoke: displays card with appropriate meta", function() {
  const caption = "Test Caption";
  const src = "testImage.png";
  const currNum = 1;
  const totalNum = 2;

  render(<Card
    caption={ caption }
    src= { src }
    currNum = { currNum }
    totalNum={ totalNum }/>);
});

it("SnapShot: displays same card as last time", function() {

  const caption = "Test Caption";
  const src = "testImage.png";
  const currNum = 1;
  const totalNum = 2;

  const { container } = render(<Card
    caption={ caption }
    src= { src }
    currNum = { currNum }
    totalNum={ totalNum }/>);

  expect(container).toMatchSnapshot();
});