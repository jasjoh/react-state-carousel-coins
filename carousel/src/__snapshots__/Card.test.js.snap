// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnapShot: displays same card as last time 1`] = `
<div>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      Test Caption
    </h4>
    <img
      alt="Test Caption"
      class="Card-image"
      src="testImage.png"
    />
    <small
      class="Card-small"
    >
      Image 
      1
       of 
      2
      .
    </small>
  </div>
</div>
`;
